python/llm/src/ipex_llm/transformers/models/utils.py

```diff
@@ -315,17 +315,23 @@ def use_esimd_sdp(q_len, k_len, head_dim, query_states):
     elif query_states.dtype != torch.float16:
         # esimd_sdp only has optimization for FP16 now
         return False
-    else:
-        device_name = torch.xpu.get_device_name(query_states.device.index)
-        if device_name.startswith("Intel(R) Arc(TM) A") or \
-                device_name.startswith("Intel(R) Data Center GPU Flex"):
-            import linear_fp16_esimd
-            if hasattr(linear_fp16_esimd, "sdp_forward"):
-                return True
-            else:
-                return False
-        else:
+    elif query_states.shape[0] > 1 and attention_mask is not None:
+        # for batched input, can't accept attention_mask
+        # TODO: this check needs some time
+        if not torch.all(attention_mask.eq(0)):
```

Comment:
hoe about this (https://discuss.pytorch.org/t/check-if-at-least-one-element-in-tensor-is-nonzero/125426/4)?

Reply:
`attention_mask.any()` is slightly faster than `not torch.all(attention_mask.eq(0))`.
Have raised a PR to update it: https://github.com/intel-analytics/ipex-llm/pull/10549